backend/baseclass.pm

```diff
@@ -51,7 +53,10 @@ __PACKAGE__->mk_accessors(
 
 sub new {
     my $class = shift;
-    my $self = bless({class => $class}, $class);
+
+    my $self = bless @_ ? @_ > 1 ? {@_} : {%{$_[0]}} : {}, $class;
```

Comment:
This is the exactly the same approach used by `Mojo::Base` to bless the new object. Basically it's taking the options from `new()`, even if they are in the array form, or hashed ones.
It makes this calls equivalent:

      myobject->new( foo => "bar", baz => "foobaz")
      myobject->new( { foo => "bar", baz => "foobaz" })

While keeping passing the same arguments to the new blessed object